jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/json/JsonCfnService.kt

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudformation.json
+
+import com.intellij.json.psi.JsonFile
+import com.intellij.json.psi.JsonObject
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import software.aws.toolkits.jetbrains.services.cloudformation.CFN_FORMAT_VERSION
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnParsedFile
+import software.aws.toolkits.jetbrains.services.cloudformation.SERVERLESS_TRANSFORM
+
+/**
+ * Acts a proxy that hides all the logic that requires JSON plugin to be enabled
+ */
+class JsonCfnService {
+    fun isCloudFormation(element: PsiElement): Boolean {
+        val psiFile = element.firstChild
+        if (psiFile !is JsonFile) {
+            return false
+        }
+
+        val fileType = psiFile.fileType
+        return if (fileType == JsonCfnFileType.INSTANCE) {
+            true
+        } else {
+            val rootElement = rootElement(psiFile)
+
+            rootElement?.findProperty(CFN_FORMAT_VERSION) != null || rootElement?.findProperty("Transform")?.value?.textMatches(SERVERLESS_TRANSFORM) == true
```

Comment:
Is this a second version of this check? I could have sworn I already saw this in the file type. Can we put the check logic into the same file even if they are not exactly the same just so there is one place for them?

Reply:
Ya, it is a slightly more encompassing check. I am not sure we need both.

Injected language support would be the only reason I could think of...but who is going to inline CFN as a String in their code 😄

Reply:
> Injected language support would be the only reason I could think of...but who is going to inline CFN as a String in their code 😄

CDK :P, makes sense we probably don't need both then

Reply:
Changed it to just file type